src/encoding.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * encoding.js
+ *
+ * This NPM module provide encoding and decoding utilities for interacting with
+ * dFusion smart contracts where manual byte encoding was needed because of
+ * solidity ABI limitations.
+ */
+import BN from "bn.js";
+import assert from "assert";
+
+export interface Order<T = string> {
+  user: string;
+  sellTokenBalance: T;
+  buyToken: number;
+  sellToken: number;
+  validFrom: number;
+  validUntil: number;
+  priceNumerator: T;
+  priceDenominator: T;
+  remainingAmount: T;
+}
+
+export interface IndexedOrder<T> extends Order<T> {
```

Comment:
It the default for `T` automatically inherited from `Order`? That is, is this line equivalent to:
```
export interface IndexedOrder<T = string> extends Order<T> {
```
?

Reply:
No, it's not. Note that you can do

```
const foo: Order | null = null;
```

but not

```
const foo: IndexedOrder | null = null;
```

The latter requires the explicit generic type. Note, that this wasn't changed or introduced in this PR.